### ISSUE TITLE
IDISP013 wrong warnings?

### ISSUE THREAD

**Issue by nulltoken** [DotNetAnalyzers/IDisposableAnalyzers#89]
The following are being frowned at by version 2.0.2.2.

Are they really relevant?

```csharp
namespace RoslynSandbox
{
    using System.Net;
    using System.Threading.Tasks;

    public class Foo
    {
        public async Task<string> Bar()
        {
            using (var client = new WebClient())
            {
                await Task.Delay(0);
                return null; // IDISP013 target
            }
        }
    }
}
```

```csharp
namespace RoslynSandbox
{
    using System.Net;
    using System.Threading.Tasks;

    public class Foo
    {
        public Task<string> Bar()
        {
            using (var client = new WebClient())
            {
                return null; // IDISP013 target
            }
        }
    }
}
```

**Comment by nulltoken** [comment]
Thanks for the quick feedback.

> maybe you can provide a more real example for when we should make exceptions?

Fair enough :wink: Below a slightly more real-world scenario.

```csharp
namespace RoslynSandbox
{
    using System;
    using System.Net;
    using System.Net.Http;
    using System.Threading.Tasks;

    public class Foo
    {
        private async Task<string> Retrieve(HttpClient client, Uri location)
        {
            using (HttpResponseMessage response = await client.GetAsync(location))
            {
                if (response.StatusCode != HttpStatusCode.OK)
                {
                    return null;
                }

                return await response.Content.ReadAsStringAsync();
            }
        }
    }
}
```

FWIW, returning `default(string)` or just `default` doesn't raise the warning.

**Comment by JohanLarsson** [comment]
Nice example, we special case return null. Do you want to have a go at it? If not I can fix it but not today, digging a well here :)

**Comment by nulltoken** [comment]
>Do you want to have a go at it?

@JohanLarsson Thanks for proposing, but unfortunately my plate is a tad full at this moment.

However, I can commit to keep on raising issues (with real-world-like repro case) whenever I bump into anything that looks weird. :wink:

**Comment by JohanLarsson** [comment]
@nulltoken Not a problem, I'll fix it when I find time, hopefully this weekend. And the repro code in the initial post was awesome just me not getting the use case.

**Comment by JohanLarsson** [comment]
https://www.nuget.org/packages/IDisposableAnalyzers/2.0.3

We need to look at the other issues next, sorry about the delay.
